core/chains/evm/logpoller/parser.go

```diff
@@ -0,0 +1,499 @@
+package logpoller
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+)
+
+const (
+	sequenceFieldAlias = "sequence"
+	sequenceField      = "CONCAT(block_number, '-', tx_hash, '-', log_index)"
+	blockFieldName     = "block_number"
+	timestampFieldName = "block_timestamp"
+	txHashFieldName    = "tx_hash"
+	eventSigFieldName  = "event_sig"
+	chainIDFieldName   = "evm_chain_id"
+)
+
+var (
+	ErrUnexpectedCursorFormat = errors.New("unexpected cursor format")
+)
+
+// pgDSLParser is a visitor that builds a postgres query and arguments from a commontypes.QueryFilter
+type pgDSLParser struct {
+	args *queryArgs
+
+	// transient properties expected to be set and reset with every expression
+	expression string
+	err        error
```

Comment:
Shouldn't we have a slice of errors here? Otherwise, every new error spotted during processing will override existing one, right?

Reply:
The parser builds SQL expressions piece by piece for each `Accept` function call and resets the error and expression values after every call. While each function could produce multiple errors, those could be satisfied the Go way with `errrors.Join`.